src/paste-markdown-html.ts

```diff
@@ -0,0 +1,84 @@
+import {insertText} from './text'
+
+export function install(el: HTMLElement): void {
+  el.addEventListener('paste', onPaste)
+}
+
+export function uninstall(el: HTMLElement): void {
+  el.removeEventListener('paste', onPaste)
+}
+
+type MarkdownTransformer = (element: HTMLElement | HTMLAnchorElement, args: string[]) => string
+
+function onPaste(event: ClipboardEvent) {
+  const transfer = event.clipboardData
+  if (!transfer || !hasHTML(transfer)) return
+
+  const field = event.currentTarget
+  if (!(field instanceof HTMLTextAreaElement)) return
+
+  // Get the plaintext and html version of clipboard contents
+  let text = transfer.getData('text/plain')
+  const textHTML = transfer.getData('text/html')
+  if (!textHTML) return
+
+  text = text.trim()
+  if (!text) return
+
+  // Generate DOM tree from HTML string
+  const parser = new DOMParser()
+  const doc = parser.parseFromString(textHTML, 'text/html')
+
+  const a = doc.getElementsByTagName('a')
+  const markdown = transform(a, text, linkify as MarkdownTransformer)
+
+  // If no changes made by transforming
+  if (markdown === text) return
+
+  event.stopPropagation()
+  event.preventDefault()
+
+  insertText(field, markdown)
+}
+
+// Build a markdown string from a DOM tree and plaintext
+function transform(
+  elements: HTMLCollectionOf<HTMLElement>,
+  text: string,
+  transformer: MarkdownTransformer,
+  ...args: string[]
+): string {
+  const markdownParts = []
+  for (const element of elements) {
+    const textContent = element.textContent || ''
+    const {part, index} = trimAfter(text, textContent)
+    markdownParts.push(part.replace(textContent, transformer(element, args)))
+    text = text.slice(index)
+  }
+  markdownParts.join(text)
```

Comment:
I think this should be `push` 🙏 

```suggestion
  markdownParts.push(text)
```